futures-channel/build.rs

```diff
@@ -0,0 +1,40 @@
+use std::env;
+
+include!("no_atomic_cas.rs");
+
+// The rustc-cfg listed below are considered public API, but it is *unstable*
+// and outside of the normal semver guarantees:
+//
+// - `futures_no_atomic_cas`
+//      Assume the target does not have atomic CAS (compare-and-swap).
+//      This is usually detected automatically by the build script, but you may
+//      need to enable it manually when building for custom targets or using
+//      non-cargo build systems that don't run the build script.
+//
+// With the exceptions mentioned above, the rustc-cfg strings below are
+// *not* public API. Please let us know by opening a GitHub issue if your build
+// environment requires some way to enable these cfgs other than by executing
+// our build script.
```

Comment:
As said in https://github.com/rust-lang/futures-rs/pull/2400#issuecomment-823391327, exposes cfg(futures_no_atomic_cas) as unstable public API (outside of the normal semver guarantees).

Reply:
Also, we probably need to set `#![cfg_attr(docsrs, doc(cfg_hide(futures_no_atomic_cas)))]` when https://github.com/rust-lang/rust/pull/79341 merged...

Reply:
That might be a while :sweat_smile: